.github/workflows/ci.yml

```diff
@@ -110,7 +110,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.10'
+          python-version: '3.11'
```

Comment:
Why not moving this version number to the env section with a variable called MAIN_PYTHON_VERSION? That way, next time you only have to modify it once rather than multiple times throughout the entire file.

Reply:
Updated as per our discussion.